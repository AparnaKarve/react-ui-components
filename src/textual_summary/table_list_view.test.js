import React from 'react';
import renderer from 'react-test-renderer';
import TableListView from './table_list_view';

describe('TableListView', () => {
  it('renders just fine...', () => {
    const tableData = {
      title: 'Tables with the Most Rows',
      headers: ['Name', 'Rows'],
      values: [
        {title: 'vim_performance_states', gname: 'vim_performance_states', value: '676,150', explorer: true, link: 'miqTreeActivateNode(\'vmdb_tree\', \'tb-10000000000181\');'},
        {title: 'miq_report_result_details', gname: 'miq_report_result_details', value: '280,848', explorer: true, link: 'miqTreeActivateNode(\'vmdb_tree\', \'tb-10000000000197\');'}
      ],
      colOrder: ['name', 'value'],
    };

    const component = renderer.create(<TableListView
      title={tableData.title}
      headers={tableData.headers}
      values={tableData.values}
      colOrder={tableData.colOrder}
    />);

    const row = component.toJSON();
    expect(row).toMatchSnapshot();
  });
});
