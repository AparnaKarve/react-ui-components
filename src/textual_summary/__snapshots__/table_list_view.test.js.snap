// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableListView renders just fine... 1`] = `
<table
  className="table table-bordered table-hover table-striped table-summary-screen"
>
  <thead>
    <tr>
      <th
        align="left"
        colSpan={2}
      >
        Tables with the Most Rows
      </th>
    </tr>
    <tr>
      <th>
        Name
      </th>
      <th>
        Rows
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      onClick={[Function]}
    >
      <td
        title="View the table"
      />
      <td
        title="View the table"
      >
        676,150
      </td>
    </tr>
    <tr
      onClick={[Function]}
    >
      <td
        title="View the table"
      />
      <td
        title="View the table"
      >
        280,848
      </td>
    </tr>
  </tbody>
</table>
`;
